From 58b64284f8e054662bd7e2edb8e19b7bdff5165a Mon Sep 17 00:00:00 2001
From: Neo Fang <neo.fang@intel.com>
Date: Fri, 17 May 2024 02:44:45 +0000
Subject: [PATCH] Change iacamera provider name in sepolicy

Issue Detailed: iacamerahal does not support AIDL which necessary in
Android 14.

Fix Detailed: change the provider name in sepolicy.

Tested-On: verified virtio camera preview/picture/record and 4 camera
preview in Intel RPL POC platform.

Tracked-On: OAM-118390
Signed-off-by: Neo Fang neo.fang@intel.com
---
 camera-ext/ivi/file_contexts    | 4 ++--
 camera-ext/ivi/service_contexts | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)
 create mode 100644 camera-ext/ivi/service_contexts

diff --git a/camera-ext/ivi/file_contexts b/camera-ext/ivi/file_contexts
index b5f44fa..fe98a4c 100644
--- a/camera-ext/ivi/file_contexts
+++ b/camera-ext/ivi/file_contexts
@@ -1,4 +1,4 @@
-/dev/media[0-9]+          u:object_r:video_device:s0
+/dev/video[0-9]+          u:object_r:video_device:s0
 /(vendor|system/vendor)/bin/android\.hardware\.automotive\.evs@1\.[0-9]-sample  u:object_r:hal_evs_default_exec:s0
 /(vendor|system/vendor)/bin/hw/android\.vendor\.hardware\.camera\.provider@2\.[0-9]+-ivi-service       u:object_r:hal_camera_default_exec:s0
-/(vendor|system/vendor)/bin/hw/android\.iacamera\.provider@2\.[0-9]+-ivi-service u:object_r:hal_camera_default_exec:s0
+/(vendor|system/vendor)/bin/hw/android\.iacamera\.provider-ivi-service u:object_r:hal_camera_default_exec:s0
diff --git a/camera-ext/ivi/service_contexts b/camera-ext/ivi/service_contexts
new file mode 100644
index 0000000..97ea8c1
--- /dev/null
+++ b/camera-ext/ivi/service_contexts
@@ -0,0 +1 @@
+android.hardware.camera.provider.ICameraProvider/ivi/0 u:object_r:hal_camera_service:s0
-- 
2.25.1

